app/controllers/TaxReturnReceivedController.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import common.SessionValues
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.AuthorisedAction
+import controllers.predicates.TaxYearAction.taxYearAction
+import javax.inject.{Inject, Singleton}
+import models.TaxReturnReceivedModel
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.AuthService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.TaxReturnReceivedView
+import utils.SessionDataHelper
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class TaxReturnReceivedController @Inject()(val authorisedAction: AuthorisedAction,
+                                            authService: AuthService,
+                                            val taxReturnReceivedView: TaxReturnReceivedView,
+                                            errorHandler: ErrorHandler,
+                                            implicit val appConfig: AppConfig,
+                                            implicit val mcc: MessagesControllerComponents,
+                                            implicit val ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with SessionDataHelper {
+
+  val timeStamp: String = DateTimeFormat.forPattern("d MMMM YYYY").print(DateTime.now(DateTimeZone.UTC))
```

Comment:
thinking about it, I think the timestamp should be in the sessionData. As here if you refreshed the page you'd get a different timestamp - and possibly different date displayed.

Reply:
Oh and probably should be UK time not UTC

Reply:
Would it not make sense to get the timestamp included as part of the model sent through from V&C?

Reply:
yeah that's what I meant by sessionData

Reply:
Ok, that will probably need to be implemented by Duane on the V&C page, afaik there's no timestamp being implemented there yet